mono/mini/mini-posix.c

```diff
@@ -315,7 +315,9 @@ static gint32 profiler_interrupt_signals_received;
 MONO_SIG_HANDLER_FUNC (static, profiler_signal_handler)
 {
 	int old_errno = errno;
-	int hp_save_index;
+
+	InterlockedWrite (&mono_thread_info_current ()->profiler_signal_ack, 1);
```

Comment:
Won't this change will cause us to stop signaling threads that have one signal discarded by the kernel?

This is a pretty common occurrence on Apple targets IIRC.